fedn/cli/client_cmd.py

```diff
@@ -65,12 +65,46 @@ def list_clients(ctx, protocol: str, host: str, port: str, token: str = None, n_
 
     try:
         response = requests.get(url, headers=headers)
-        print_response(response, "clients")
+        print_response(response, "clients", None)
     except requests.exceptions.ConnectionError:
```

Comment:
We should handle more than just ConnectionError, there are plenty of other error related to http requests. You don't have do do this know but make an other issue of it. For example create a utility function that handles all requests and checks the response for error:


import requests
from requests.exceptions import HTTPError, Timeout, ConnectionError, RequestException

def make_request(method, url, **kwargs):
    """
    A reusable function to handle HTTP(S) requests with error handling.

    Args:
        method (str): HTTP method (e.g., 'GET', 'POST').
        url (str): The URL for the request.
        **kwargs: Additional arguments passed to `requests.request()`.

    Returns:
        dict: A dictionary with the response data or an error message.
    """
    try:
        response = requests.request(method, url, **kwargs)
        response.raise_for_status()
        # Assuming JSON response, adjust if needed
        return {"success": True, "data": response.json()}
    except HTTPError as http_err:
        return {"success": False, "error": f"HTTP error occurred: {http_err}"}
    except Timeout:
        return {"success": False, "error": "The request timed out"}
    except ConnectionError:
        return {"success": False, "error": "Connection error occurred"}
    except RequestException as req_err:
        return {"success": False, "error": f"Request error: {req_err}"}
    except Exception as e:
        return {"success": False, "error": f"Unexpected error: {e}"}